doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -0,0 +1,97 @@
+.. _ivshmem-hld:
+
+ACRN Shared Memory Based Inter-VM Communication
+###############################################
+
+To support inter virtual machines communication based on shared memory
+mechanism, ACRN device model/hypervisor emulate a virtual PCI device
+(Called ivshmem device) to expose the shared memory base address and
+size. Currently, ACRN only supports post-launched to post-launched
+communication via ivshmem device. In future, ACRN will supports
+pre-launched VM inter-VM communication.
+
+Inter-VM Communication Overview
+*******************************
+
+.. figure:: images/ivshmem-image1.png
+   :align: center
+   :name: ivshmem-architecture-overview
+
+   ACRN Inter-VM Communication Architecture Based on Shared Memory
+
+Two solutions for ivshmem device implementation, one is called DM-land,
+another is called Hypervisor-land. For DM-land, the ivshmem device is
+emulated in device model and the inter-VM communication shared memory
+region is allocated from Service memory. For Hypervisor-land, the
+ivshmem device is emulated in hypervisor and the shared memory is
+allocated from hypervisor. The former only work for post-launched VMs,
+the latter can work for both of pre-launched and post-launched VMs. The
+Hypervisor-land will be implemented in the future.
+
+-  Ivshmem dm – The ivshmem device model implements device related
+   registers and shared memory mapping. In the future it will also
+   support the interrupt mechanism.
+
+-  Ivshmem server – A daemon for inter-VM notification capability,
+   currently, it has not been implemented, so the inter-VM communication
+   doesn’t support notification mechanism.
+
+Ivshmem Device Introduction
+***************************
+
+The ivshmem device is a virtual standard PCI device, it consists of two
+BARs(BAR0 and BAR2), one is used for interrupt related register and BAR
+size is 256 bytes, another is used for exposing shared memory base
```

Comment:
one is used for emulate interrupt related registers, another one is used for exposing shared memory region. The ivshmem device doesn't support any extra capabilities.